### DISCUSSION TITLE
Conflict resolution by renaming in reverse

### DISCUSSION THREAD

**Discussion by mavhc** [ayoisaiah/f2#91] [Q&A]
Give a list of files
file01
file02
file03

and a wish to increase the number on each by 1

file02
file03
file04

using

f2 -f "file\d\d" -r "file{2%02d}" -l 1

it fails due to conflicts, it can't rename file01 to file02 because file02 exists

However this would work if file03 was first renamed to file04

Is this possible to do now/add as a feature?

**Comment by ayoisaiah** [comment]
You can [index with a capture variable](https://f2.freshman.tech/guide/indexing.html#_7-indexing-using-a-capture-variable) and [sort it](https://f2.freshman.tech/guide/sorting.html) naturally in reverse to achieve what you want:

```bash
f2 -f 'file(\d+)' -r 'file{$1%02d1}' -sortr natural
```

![image](https://github.com/user-attachments/assets/ff1f0de3-783c-4bbb-b161-01fb5c3d351c)
